cmake/NeuronFileLists.cmake

```diff
@@ -218,6 +218,11 @@ set(NRNIV_FILE_LIST
     netpar.cpp
     nonlinz.cpp
     nrnbbcore_write.cpp
+    nrnbbcore_write/callbacks/bbcore_callbacks.cpp
+    nrnbbcore_write/data/cell_group.cpp
+    nrnbbcore_write/data/datum_indices.cpp
+    nrnbbcore_write/io/bbcore_write.cpp
+    nrnbbcore_write/utils/corenrn_utils.cpp
```

Comment:
As Michael was saying, we can rename `nrnbbcore_write` folder.

@nrnhines : should we add this as corewrite or something else?

Reply:
I like the token nrncore in place of any nrnbbcore at least when it belongs to NEURON and especially if it suggests a
neuron -> coreneuron direction. I suppose that with respect to variable and array names one could use nrn2core and core2nrn if it adds clarity. But I'm willing to entertain other ideas.

Reply:
`nrncore` sounds good to me.